drivers/staging/vc04_services/bcm2835-isp/bcm2835-v4l2-isp.c

```diff
@@ -1760,63 +1757,61 @@ static int bcm2835_isp_probe_instance(struct platform_device *pdev,
 	return 0;
 }
 
-static int bcm2835_isp_remove(struct platform_device *pdev)
+static void bcm2835_isp_remove(struct vchiq_device *device)
 {
 	struct bcm2835_isp_dev **bcm2835_isp_instances;
 	unsigned int i;
 
-	bcm2835_isp_instances = platform_get_drvdata(pdev);
+	bcm2835_isp_instances = vchiq_get_drvdata(device);
 	for (i = 0; i < BCM2835_ISP_NUM_INSTANCES; i++) {
 		if (bcm2835_isp_instances[i])
 			bcm2835_isp_remove_instance(bcm2835_isp_instances[i]);
 	}
-
-	return 0;
 }
 
-static int bcm2835_isp_probe(struct platform_device *pdev)
+static int bcm2835_isp_probe(struct vchiq_device *device)
 {
 	struct bcm2835_isp_dev **bcm2835_isp_instances;
 	unsigned int i;
 	int ret;
 
-	bcm2835_isp_instances = devm_kzalloc(&pdev->dev,
+	bcm2835_isp_instances = devm_kzalloc(&device->dev,
 					     sizeof(bcm2835_isp_instances) *
 						      BCM2835_ISP_NUM_INSTANCES,
 					     GFP_KERNEL);
 	if (!bcm2835_isp_instances)
 		return -ENOMEM;
 
-	platform_set_drvdata(pdev, bcm2835_isp_instances);
+	vchiq_set_drvdata(device, bcm2835_isp_instances);
 
 	for (i = 0; i < BCM2835_ISP_NUM_INSTANCES; i++) {
-		ret = bcm2835_isp_probe_instance(pdev,
+		ret = bcm2835_isp_probe_instance(device,
 						 &bcm2835_isp_instances[i], i);
 		if (ret)
 			goto error;
 	}
 
-	dev_info(&pdev->dev, "Loaded V4L2 %s\n", BCM2835_ISP_NAME);
+	dev_info(&device->dev, "Loaded V4L2 %s\n", BCM2835_ISP_NAME);
 	return 0;
 
 error:
-	bcm2835_isp_remove(pdev);
+	bcm2835_isp_remove(device);
 
 	return ret;
 }
 
-static struct platform_driver bcm2835_isp_pdrv = {
+static struct vchiq_driver bcm2835_isp_drv = {
 	.probe = bcm2835_isp_probe,
 	.remove = bcm2835_isp_remove,
 	.driver = {
 			.name = BCM2835_ISP_NAME,
 		  },
 };
 
-module_platform_driver(bcm2835_isp_pdrv);
+module_vchiq_driver(bcm2835_isp_drv);
 
 MODULE_DESCRIPTION("BCM2835 ISP driver");
 MODULE_AUTHOR("Naushir Patuck <naush@raspberrypi.com>");
 MODULE_LICENSE("GPL");
 MODULE_VERSION("1.0");
-MODULE_ALIAS("platform:bcm2835-isp");
+MODULE_ALIAS("vchiq:bcm2835-isp");
```

Comment:
Argh, this is inconsistent compared to bcm2835-codec. And also I think the upstream(/staging) drivers remove this in favour of a compatible ID table that should also be adapted in here.